src/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,318 @@
+    uint256 public minCollateralizationRatio;
+
+    /// @notice Total debt taken out
+    uint256 public debtBalance;
+
+    /// @notice Fixed borrow interest rate in bpds
+    uint256 public immutable interestRateBps;
+
+    /// @notice Amount in seconds for interest to accumulate
+    uint256 public immutable interestRatePeriod;
+
+    /// @notice Address to send bad debt collateral
+    address public debtCollector;
+
+    /// Debt parameters
+
+    /// @notice Debt token address
+    IERC20 public immutable debtToken;
+
+    /// @notice Debt token price
+    uint256 public debtPrice;
+    
+    /// @notice Mapping of user positions
+    mapping(address => Position) public positions;
+
+    event DepositDebt(uint256 amount);
+    event RemoveDebt(uint256 amount);
+    event PostCollateral(address account, uint256 amount);
+    event Borrow(address account, uint256 amount);
+    event Repay(address account, uint256 amount);
+    event Withdraw(address account, uint256 amount);
+    event Liquidated(address account, uint256 debtAmount, uint256 collateralSeized);
+
+    error InvalidAmount(uint256 amount);
+    error InsufficentCollateral(uint256 maxCapacity, uint256 debtAmount);
+    error ExceededBorrowedAmount(address account, uint256 amountBorrowed, uint256 amountRepay);
+    error ExceededCollateralAmonut(address account, uint256 amountCollateral, uint256 collateralWithdraw);
+    error WillUnderCollaterlize(address account, uint256 withdrawalAmount);
+    error OverCollaterilized(address account);
+    
+    constructor(
+        uint256 _interestRateBps,
+        uint256 _minCollateralizationRatio,
+        uint256 _interestRatePeriod,
+
+        address _collateralToken,
+        uint256 _collateralPrice,
+
+        address _debtToken,
+        uint256 _debtPrice,
+        address _debtCollector
+
+    ) {
+        interestRateBps = _interestRateBps;
+        minCollateralizationRatio = _minCollateralizationRatio;
+        interestRatePeriod = _interestRatePeriod;
+        
+        collateralToken = IERC20(_collateralToken);
+        collateralPrice = _collateralPrice;
+
+        debtToken = IERC20(_debtToken);
+        debtPrice = _debtPrice;
+        
+        debtCollector = _debtCollector;
+    }
+
+
+    function addOperator(address _address) external override onlyOwner {
+        _addOperator(_address);
+    }
+
+    function removeOperator(address _address) external override onlyOwner {
+        _removeOperator(_address);
+    }
+
+    function setDebtPrice(uint256 _debtPrice) external onlyOperators {
+        debtPrice = _debtPrice;
+    }
+
+    function setCollateralPrice(uint256 _collateralPrice) external onlyOperators {
+        collateralPrice = _collateralPrice;
+    }
+
+    function setDebtCollector(address _debtCollector) external onlyOperators {
+        debtCollector = _debtCollector;
+    }
+
+    function setCollateralizationRatio(uint256 _minCollateralizationRatio) external onlyOperators {
```

Comment:
Should rename to `setMinCollateralizationRatio()` to be consistent with param name